lib/valkyrie/resource.rb

```diff
@@ -133,5 +190,20 @@ def to_s
     def human_readable_type
       self.class.human_readable_type
     end
+
+    def [](name)
+      unless name.is_a?(Symbol)
+        warn "[DEPRECATION] #[] only accepts symbol arguments in dry-struct now. Support for string arguments " \
+             "will be removed in the next major release of Valkyrie. Please use symbols. " \
+             "Called from #{Gem.location_of_caller.join(':')}"
+      end
+      super(name.to_sym)
+    rescue NoMethodError
+      nil
+    end
+
+    def set_value(key, value)
+      instance_variable_set(:"@#{key}", self.class.schema[key.to_sym].call(value))
```

Comment:
Should there be a deprecation warning here if they are setting with a String?

Reply:
That's an interesting question. This is our own API. `to_sym` is so easy I'm inclined to let them pass a string here. What do you think? Is it confusing to only allow strings here?

Reply:
Yeah, okay. I think the question comes up too, if they pass a string should we be erroring at them in `2.0`? If not, I think it'll just...not work

Reply:
@cam156 Can you take another look? I've changed my tact with this - both `#[]` and `#set_value` accept strings now.